Libraries/Opc.Ua.PubSub/Uadp/UadpDataSetMessage.cs

```diff
@@ -0,0 +1,957 @@
+                            break;
+                        case BuiltInType.ExpandedNodeId:
+                            binaryEncoder.WriteExpandedNodeIdArray("ExpandedNodeIdArray", (ExpandedNodeId[])variant.Value);
+                            break;
+                        case BuiltInType.StatusCode:
+                            binaryEncoder.WriteStatusCodeArray("StatusCodeArray", (StatusCode[])variant.Value);
+                            break;
+                        case BuiltInType.XmlElement:
+                            binaryEncoder.WriteXmlElementArray("XmlElementArray", (System.Xml.XmlElement[])variant.Value);
+                            break;
+                        case BuiltInType.Variant:
+                            binaryEncoder.WriteVariantArray("VariantArray", (Variant[])variant.Value);
+                            break;
+                        case BuiltInType.Enumeration:
+                            //TODO make this work
+                            //binaryEncoder.WriteInt32Array("EnumerationArray", Convert.ToInt32(variant.Value));
+                            binaryEncoder.WriteVariantArray("EnumerationArray", (Variant[])variant.Value);
+                            break;
+                        case BuiltInType.ExtensionObject:
+                            binaryEncoder.WriteExtensionObjectArray("ExtensionObjectArray", (ExtensionObject[])variant.Value);
+                            break;
+                    }
+                }
+            }
+            catch(Exception ex)
+            {
+                Utils.Trace("Error encoding field {0} - {1}", field.FieldMetaData.Name, ex);
+            }
+        }
+
+        /// <summary>
+        /// Decode RawData type (for SimpleTypeDescription!?)
+        /// </summary>
+        /// <param name="binaryDecoder"></param>
+        /// <param name="fieldMetaData"></param>
+        /// <returns></returns>
+        private object DecodeRawData(BinaryDecoder binaryDecoder, FieldMetaData fieldMetaData)
+        {
+            if (fieldMetaData.BuiltInType != 0)// && fieldMetaData.DataType.Equals(new NodeId(fieldMetaData.BuiltInType)))
+            {
+                try
+                {
+                    switch (fieldMetaData.ValueRank)
+                    {
+                       
+                        case ValueRanks.Scalar:
+                            return DecodeRawScalar(binaryDecoder, fieldMetaData.BuiltInType);
+                       
+                        case ValueRanks.OneDimension:
+                            return DecodeRawArrayOneDimension(binaryDecoder, (BuiltInType)fieldMetaData.BuiltInType);
+
+                        case ValueRanks.TwoDimensions:
+                        case ValueRanks.OneOrMoreDimensions:
+                            //return DecodeRawArrayMultiDimension(binaryDecoder, (BuiltInType)fieldMetaData.BuiltInType, fieldMetaData.ArrayDimensions);
```

Comment:
what happens on yet unsupported array dimensions?

Reply:
only decode for Scalar and OneDimension fileds yet..